src/pint/utils.py

```diff
@@ -434,3 +435,113 @@ def show_param_cov_matrix(matrix, params, name="Covaraince Matrix", switchRD=Fal
     contents = output.getvalue()
     output.close()
     return contents
+
+
+def dmxparse(fitter, save=False):
+    """Function to run dmxparse in python using PINT objects and results.
+    Based off of dmxparse by Paul Demorest (https://github.com/nanograv/tempo/tree/master/util/dmxparse)
+    
+    Inputs:
+    :param fitter: PINT fitter used to get timing residuals, must have already run GLS fit
+    :param save: boolean; if True saves output to text file in the format used for the TEMPO version. 
+          If not output save file is desired, save = False (which is the default)
+          Output file name is dmxparse.out"""
+    # We get the DMX values, errors, and mjds (same as in getting the DMX values for DMX v. time)
+    # Get number of DMX epochs
+    dmx_epochs = 0
+    for p in fitter.model.params:
+        if "DMX_" in p:
+            dmx_epochs += 1
+    # Get DMX values (will be in units of 10^-3 pc cm^-3)
+    DMX_keys = []
+    DMXs = []
+    DMX_Errs = []
+    DMX_R1 = []
+    DMX_R2 = []
+    DMX_center_MJD = []
+    for ii in range(dmx_epochs):
+        ii += 1
```

Comment:
This is kind of a dangerous line here since ii is already being defined in the range() loop above.  If you want ii to start with 1, then do for ii in range(1,dmx_epochs+1)

Reply:
Thanks Scott, I'll fix that up too and generally try to fix it up if there are other issues.